### PR TITLE
Remove QuickCheck from production code and add missing module descriptions

### PR DESCRIPTION
**Pull request by larskuhtz**

### CODE REVIEW

File: src/Chainweb/HostAddress.hs [hunk 1]
```diff
@@ -232,7 +219,7 @@ ipV6Parser = p0
 
 portParser :: Parser Port
 portParser = Port
-    <$> (decimal >>= \(d :: Integer) -> int d <$ guard (d < 2^(16 :: Int)))
+    <$> (decimal >>= \(d :: Integer) -> int d <$ guard (d <= 2^(16 :: Int)-1))
```
Comment (emilypi) [inline src/Chainweb/HostAddress.hs:222]: isn't this an equivalent statement?

File: src/Chainweb/Pact/NoCoinbase.hs [hunk 2]
```diff
@@ -8,7 +8,7 @@
 -- Maintainer: Lars Kuhtz <lars@kadena.io>
 -- Stability: experimental
 --
--- TODO
+-- A fake coin base for testing purposes.
```
Comment (emilypi) [inline src/Chainweb/Pact/NoCoinbase.hs:11]: It's not fake - it's necessary for genesis transactions

File: src/Chainweb/HostAddress.hs (hunk 1 above)
Comment (larskuhtz) [reply src/Chainweb/HostAddress.hs:222]: it is for `d :: Integer`. I got bitten by copy and pasting it to a context with `d :: Word16` in which case it isn't equivalent. So, I thought it was better to use the more robust code.

Comment (emilypi) [reply src/Chainweb/HostAddress.hs:222]: ahh interesting alright!

File: src/Chainweb/Pact/NoCoinbase.hs (hunk 2 above)
Comment (larskuhtz) [inline src/Chainweb/Pact/NoCoinbase.hs:11]:
```suggestion
-- A noop coin base for genesis transactions and testing purposes.
```

Comment (larskuhtz) [reply src/Chainweb/Pact/NoCoinbase.hs]: Oh, I didn't know that. I fixed the comment.

**Pull request merged**